ctapipe/plotting/charge_resolution.py

```diff
@@ -0,0 +1,360 @@
+
+        Also contains the equation from which the charge resolution
+        requirement is defined, which can be plotted alongside the charge
+        resolution curve.
+
+        `ctapipe.tools.plot_charge_resolution` demonstrated the use of this
+        component.
+
+        Parameters
+        ----------
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        tool : ctapipe.core.Tool
+            Tool executable that is calling this component.
+            Passes the correct logger to the component.
+            Set to None if no Tool to pass.
+        kwargs
+        """
+        super().__init__(config=config, parent=tool, **kwargs)
+        self._df_pixel = None
+        self._df_camera = None
+
+        self.fig = plt.figure(figsize=(12, 7.42))
+        self.ax = self.fig.add_subplot(111)
+
+        self.ax.set_xlabel("True Charge (p.e.)")
+        self.ax.set_ylabel(
+            r"Fractional Charge Resolution $\frac{{\sigma_Q}}{{Q}}$"
+        )
+
+        if not self.output_path:
+            raise ValueError("Output path must be specified")
+
+    def _set_file(self, path):
+        """
+        Prepare the class to plot from a particular charge resolution file.
+
+        Parameters
+        ----------
+        path : str
+            Path to the charge resolution HDF5 file
+        """
+        with pd.HDFStore(path, 'r') as store:
+            self._df_pixel = store['charge_resolution_pixel']
+            self._df_camera = store['charge_resolution_camera']
+
+    def _plot(self, x, y, xerr, yerr, label=''):
+        """
+        Plot the given points onto the figure
+
+        Parameters
+        ----------
+        x : ndarray
+        y : ndarray
+        xerr : ndarray
+        yerr : ndarray
+        label : str
+        """
+        (_, caps, _) = self.ax.errorbar(
+            x, y, xerr=xerr, yerr=yerr, mew=1, capsize=1, elinewidth=0.5,
+            markersize=2, label=label, linewidth=0.5, fmt='.',
+        )
+        for cap in caps:
+            cap.set_markeredgewidth(0.5)
+
+    def plot_average(self, path, label=''):
+        """
+        Plot the average and standard deviation of the charge resolution
+        across the pixels of the camera.
+
+        Parameters
+        ----------
+        path : str
+            Path to the charge resolution HDF5 file
+        label : str
+            Label for the figure's legend
+        """
+        self._set_file(path)
+        df_binned = bin_dataframe(self._df_pixel, self.n_bins)
+        agg = {'charge_resolution': ['mean', 'std'], 'true': 'mean'}
+        df_agg = df_binned.groupby(['bin']).agg(agg)
+        x = df_agg['true']['mean'].values
+        y = df_agg['charge_resolution']['mean'].values
+        yerr = df_agg['charge_resolution']['std'].values
+        self._plot(x, y, None, yerr, label)
```

Comment:
I believe explicitly providing `None` for the 3rd argument is very C-like, while in Python one would rather write this like:
`self._plot(x,y, yerr=yerr, label=label)`

This `foo=foo` might feel a bit repetitive, therefore some people might write this part as:
```python
self._plot(
    x=df_agg['true']['mean'].values,
    y=df_agg['charge_resolution']['mean'].values,
    yerr=df_agg['charge_resolution']['std'].values,
    label=label,
)
```

Reply:
Okay I've tried to address this a little in my latest commit